src/components/ResourceList/ResourceList.tsx

```diff
@@ -589,39 +589,42 @@ export const ResourceList: ResourceListType = function ResourceList<ItemType>({
 
   const showEmptyState = filterControl && !itemsExist && !loading;
 
-  const StickyHeader = (isSticky: boolean) => {
-    const headerClassName = classNames(
-      styles.HeaderWrapper,
-      sortOptions &&
-        sortOptions.length > 0 &&
-        !alternateTool &&
-        styles['HeaderWrapper-hasSort'],
-      alternateTool && styles['HeaderWrapper-hasAlternateTool'],
-      isSelectable && styles['HeaderWrapper-hasSelect'],
-      loading && styles['HeaderWrapper-disabled'],
-      isSelectable && selectMode && styles['HeaderWrapper-inSelectMode'],
-      isSticky && styles['HeaderWrapper-isSticky'],
-    );
-    return (
-      <div className={headerClassName} testID="ResourceList-Header">
-        <EventListener event="resize" handler={handleResize} />
-        {headerWrapperOverlay}
-        <div className={styles.HeaderContentWrapper}>
-          {headerTitleMarkup}
-          {checkableButtonMarkup}
-          {alternateToolMarkup}
-          {sortingSelectMarkup}
-          {selectButtonMarkup}
-        </div>
-        {bulkActionsMarkup}
-      </div>
-    );
-  };
   const headerMarkup = !showEmptyState &&
     (showHeader || needsHeader) &&
     listRef.current && (
       <div className={styles.HeaderOuterWrapper}>
-        <Sticky boundingElement={listRef.current}>{StickyHeader}</Sticky>
+        <Sticky boundingElement={listRef.current}>
+          {(isSticky: boolean) => {
```

Comment:
This being inlined makes it really hard to read this section. Why did you do it this way?

Reply:
I can break it back out. The check to items exist and list ref just seemed redundant as the check to items exist is what is setting the listRef in the first place. Then because we are not rendering it in two different places breaking it out seemed unnecessary. This was the only thing that I could get running locally that fixed the sticky header.

Reply:
Yeah that makes sense, but did this mean you need a way to force an update and that's why you added the reducer? Or was that because of something else?

Reply:
Exactly, yeah. This is the way it was done in the old class component so I figured we could just keep that the same way. The reducer is the way the react docs recommend to force an update.

Reply:
@AndrewMusgrave this makes sense to me, do you have any thoughts?

Reply:
@amrocha are your changes to ResourceList going to end up being pretty big? Or are these merge conflicts we will most likely be able to work through? I was trying to make a big effort to get this fixed up this weekend to help you out.

Reply:
They're fairly small: https://github.com/Shopify/polaris-react/pull/2971

I'll open a PR on your branch resolving any conflicts as soon as it's merged, so you don't have to worry about it. Personally I think your PR is good to go, but I don't work on components super often so I want to get at least one more set of eyes on it before merging. I'll get someone to do it asap so we can hopefully get this out by the end of the week.

Thanks for taking a look at this, we really appreciate it!

Reply:
Cool, thanks for the heads up. I'll familiarize myself and prep for any conflicts.